crates/blockchain/fork/src/lib.rs

```diff
@@ -988,9 +996,10 @@ impl<
             last_header.state_root,
             previous_total_difficulty,
             BlockConfig {
-                base_fee_params: &self.base_fee_params,
+                base_fee_params: self.base_fee_params.clone(),
```

Comment:
Question: Can this clone be expensive for dynamic base fee parameters? The same for the `scheduled_blob_parameters`?

If so, would it be an idea to have two types of `BlockConfig`? One that's stored and one that has references? E.g. `BlockConfigRef`? You could easily implement `AsRef<BlockConfigRef>` for `BlockConfig` or as a member function.

Reply:
Our rule of thumb is to pass anything larger than a pointer (8 bytes) by reference, if possible, so that feels like a non-trivial amount especially for the calls that repeat more often.

I'd mark it as a nice-to-have. Only implement if it's a trivial amount of work